### DISCUSSION TITLE
gold+ upgrading system rework

### DISCUSSION THREAD

**Discussion by vlad-ultramarine** [PavelZinchenko/event-horizon-frontier-builds#44] [General]
смысл новой системы улучшения модулей мне понятен, но что будет с модулями которые были улучшены по старой системе? есть игроки у которых все или большинство имеющихся в инвентаре модулей давно улучшены в +20. Эти улучшения перейдут в полностью прокачанные характеристики модулей по новой системе или что нибудь другое?

**Comment by Inedilarion** [comment]
Было бы очень хорошо перевести модули +20 в улучшенные по новой системе. Или хотя бы сбросить все +20, вернув использованные для их улучшения чертежи и деньги 
It would be very good to convert +20 modules into improved ones using the new system. Or at least reset all +20, returning the drawings and money used to improve them

**Comment by PavelZinchenko** [comment]
I didn't like the old module upgrade system from the very beginning. I wanted to create a system similar to the one in "Space Arena". The +X modules will work, but the bonuses won't stack. The largest of the bonuses will be used. I haven't decided what to do with the +X modules yet. Suggestions are welcome. Well, this is beta, everything can still change.

**Reply by Inedilarion** [reply]
Seems like +20 and new upgrades stacking now
![image](https://github.com/PavelZinchenko/event-horizon-frontier-builds/assets/66635560/791a9f02-22e4-4514-b9e9-fb7c7b1a6de9)

**Comment by Inedilarion** [comment]
В случае если новые улучшения позволяют довести уровень усиления модулей до +20 (урон и броню/щит), то использование игрой наибольшего бонуса сделает +20 автоматически устаревшим, и ничего менять не нужно.
Если предел улучшения урона и брони у нового способа меньше, то оставлять модули +20 эксклюзивными нельзя, это даст всем старым игрокам большое преимущество и будет плохо встречено всеми игроками вообще (те кто знают, не будут обновляться пока не накопят достаточно модулей +20 всех видов которые хотят, а кто не знал и обновился, будут расстроены что этого не сделали). То есть, в идеале стоит или оставлять доступным улучшение до +20, или вообще убирать все +20 из игры (желательно выдав за них компенсацию, потому что чертежей и кредитов на их производство было потрачено очень много. А то и звёзд, в случае луча Тартаруса)
P.S. Можно ли добавить в новую систему улучшений возможность временно отключать их? Модули +20 не влияли на строение сборок, а новые улучшения приведут к тому, что работающая у одного человека сборка не будет работать у другого, не имеющего достаточно улучшенных модулей (например, энергия будет уходить в минус). А с отключением можно будет собирать универсальные сборки, работающие у всех вне зависимости от прокачки

If new improvements allow player to increase the module enhancement level to +20 (damage and armor/shield), then using the largest bonus by the game will make +20 automatically obsolete, and nothing needs to be changed.
If the limit for improving damage and armor of the new method is less, then you cannot leave +20 modules exclusive, this will give all old players a big advantage and will be poorly received by all players (those who know will not update until they have accumulated enough +20 modules of all types that they want, and those who didn’t know and updated will be upset that they didn’t do it). That is, ideally, it’s worth either leaving the improvement to +20 available, or removing all +20 from the game (preferably by issuing compensation for them, because a lot of drawings and credits were spent on their production. And even stars, in the case of Tartarus beam)
P.S. Is it possible to add to the new upgrade system the ability to temporarily disable them? +20 modules did not affect the structure of the ship builds, but new improvements will lead to the fact that the build that works for one person will not work for another who does not have sufficiently improved modules (for example, the energy will go into minus). And with the switch off, it will be possible to assemble universal builds that work for everyone, regardless of leveling

**Comment by ryan1ryanryan** [comment]
Simply lowering the modules level to 5 and returning the upgrade cost from level 6 to 20 in stars would open up the option for veterans to upgrade future stuff faster, and newer players to not be locked out of anything exclusive. Of course since upgrading something like the heavy energy wave from +19 to +20 would cost a lot of stars whereas those with blueprints only had to spend credits, maybe returning just half the cost would be a decent compensation enough.

**Comment by PavelZinchenko** [comment]
There's another question bothering me, so I'd like to know your opinion. The new upgrades require a lot of resources that can be found mostly in exploration. I doubt players will enjoy grinding exploration levels. Could you suggest a better way to earn them?

**Reply by Inedilarion** [reply]
Я бы предложил нечто вроде дуэлей, или захвата баз в оригинальной игре. Против действительно мощных врагов. Потому что во фронтире после получения титанов волны начинают проходиться нажатием на кнопку старта, на арене любой враг кроме читеров разваливается от попадания био-плазменной торпеды в базу, а исследования после появления модификации корабля на 100 жизней тоже не представляют никакой сложности. И становится просто скучно. Так что хотелось бы получить какие-то действительно очень сложные активности, непроходимые полностью с теми сборками/флотами которые используются обычно. Нечто подобное есть в выживаниях, но за них не даётся ощутимой награды и они зачастую как и волны оказываются очень простыми в прохождении

I would suggest something like duels, or capturing bases in the original game. Against really powerful enemies. Because in the frontier, after receiving titans, the waves begin to pass by pressing the start button, in the arena, any enemy except cheaters falls apart when a bio-plasma torpedo hits its base, and exploration after the appearance of a ship modification for 100 lives is also quite easy. And game just gets boring. So I would like to get some really very hard activities that are completely impenetrable with the builds/fleets that are usually used. There is something similar in survivals, but there is no tangible reward for them and they, like waves, often turn out to be very easy to complete

**Reply by ryan1ryanryan** [reply]
As an addition to Inedilarion's comment, as there is a surplus of skill points that has yet to serve a function, maybe add a bonus to increase the chances of said resources to drop/ a surveyor with a cooldown that highlights nodes that'll drop the desired resources/mine the resource over X levels completed. Not really related but since you're removing rear admiral, this could be an opportunity to add a separate  pilots section that you can only have two of but have special perks, like a squid that allows 360° aim at the cost of fire rate or a robot that deals more damage at the cost of less health or a merchant that weakens the ship overall but allows the resources to drop anywhere.

**Reply by PavelZinchenko** [reply]
> So I would like to get some really very hard activities that are completely impenetrable with the builds/fleets that are usually used

You asked for it 😈

**Reply by vlad-ultramarine** [reply]
💀

**Comment by vlad-ultramarine** [comment]
I thought about the option of getting them for killing opponents in levels, but it's too easy. To make it possible for a scientist to recycle modules into resources, and leave the opportunity to receive these resources in exploration

**Reply by Zen9890** [reply]
In this case recycling some abundant stuff like armored energy cells seems to just remove grind altogether, with this system i think there will be always at least one abundant component with desired resource (intended or not)

**Comment by vlad-ultramarine** [comment]
я обдумал свое предложение насчет системы переработки модулей и пришел к выводу что наилучшим вариантом будет добавление ресурсов из Event horizon рпг, а именно полимеры, промышленные металлы, драгоценные металлы, драгоценные камни и токсичные отходы. Эти ресурсы нужно задействовать в новой системе улучшения и добавить шанс их выпадения при разборке модулей, также у каждого модуля должны быть свои ресурсы которые могут с него выпасть (например: с бронеплиты можно получить с определеным шансом промышленные металлы, обломки или только обломки которые будут сразу продаваться, с камня бесконечности можно получить, обломки, драгоценные металлы/камни, эссенцию пустоты и тд). Конечно можно сделать так чтобы не все ресурсы могли быть получены через переработку

I thought about my suggestion for a module recycling system and came to the conclusion that the best option would be to add resources from Event horizon rpg, namely polymers, industrial metals, precious metals, precious stones and toxic waste. These resources need to be used in a new improvement system and add the chance of their falling out when disassembling modules, also each module should have its own resources that can fall out of it (for example: from an armored plate you can get industrial metals, scraps or only scraps that will be immediately sold, from an infinity stone you can get scraps, precious metals/stones, void essence, etc.). Of course, it is possible to make sure that not all resources can be obtained through recycling

**Comment by vlad-ultramarine** [comment]
также можно добавить новый скилл связанный с учёным, поскольку у меня есть более 600 неиспользованных очков скиллов

**Comment by Thirteen0130** [comment]
Как насчёт не убирать систему +X, а просто сделать её дороже? Например, за +1 нужно 10 звёзд, за +2 нужно 20,+3-50 звёзд, +4-100 и тд. Новая система прокачки довольно дорогая, и постоянно ходить в разведку для получения новых предметов/артефактов, только чтобы улучшить один раз не очень хорошо. Чтобы скрасить разведку, можно добавить прожорливый крупный (9 клеток) модуль на ускорение времени исследования(+50%), но только один модуль на корабль. Я лично не хочу, чтобы убирали систему +X

**Reply by vlad-ultramarine** [reply]
новая система лучше +Х и то что ты предлагаешь вообще не по карману обычным игрокам, вот кто давно ушел за 600+ уровни и старые игроки которым компенсировали изменения древних версий звёздами конечно смогут себе это позволить

**Reply by Thirteen0130** [reply]
Так на это и расчёт. Если сделать это очень дорогим, то и убирать не надо, и некий баланс будет. Просто я полетал в 5 системах и смог улучшить лишь два модуля на пару пунктов. Это очень раздражает. Всю игру только и летать на иследования? Неееее. Да и жаль тех потраченных звёзд, ведь если +X уберут, то и уберут все прокаченные модули
